File: library/src/main/java/org/crispy_fish/filetype/ecf/EventControlFileAssistant.java
package org.crispy_fish.filetype.ecf;

import java.nio.file.FileSystems;
import java.nio.file.Path;
import java.nio.file.PathMatcher;

public class EventControlFileAssistant {

    private PathMatcher eventControlFileMatcher;

    public EventControlFileAssistant() {
        // no-op
    }

    public boolean isEventControlFilePath(Path path) {
        return getEventControlFileMatcher().matches(path.getFileName())
                && !path.toFile().isDirectory();
    }

    private synchronized PathMatcher getEventControlFileMatcher() {
        if (eventControlFileMatcher == null) {
            eventControlFileMatcher = buildEventControlFileMatcher();
        }
        return eventControlFileMatcher;
    }

    PathMatcher buildEventControlFileMatcher() {
        return FileSystems.getDefault().getPathMatcher("glob:*.ecf");
    }
}

File: library/src/main/kotlin/org/coner/crispy_fish/filetype/ecf/EventControlFileAssistant.kt
package org.coner.crispy_fish.filetype.ecf

import java.nio.file.FileSystems
import java.nio.file.Path
import java.nio.file.PathMatcher

class EventControlFileAssistant {

    private val eventControlFileMatcher: PathMatcher by lazy {
        FileSystems.getDefault().getPathMatcher("glob:*.ecf")
    }

    fun isEventControlFilePath(path: Path): Boolean {
        return eventControlFileMatcher.matches(path.fileName) && !path.toFile().isDirectory
    }

}
